lib/test-utils/src/Test/QuickCheck/Extra.hs

```diff
@@ -302,6 +314,22 @@ shrinkInterleaved (a, shrinkA) (b, shrinkB) = interleave
     interleave xs [] = xs
     interleave [] ys = ys
 
+--------------------------------------------------------------------------------
+-- Generating and shrinking natural numbers
+--------------------------------------------------------------------------------
+
+chooseNatural :: (Natural, Natural) -> Gen Natural
+chooseNatural (lo, hi) =
+    choose (intCast lo, intCast hi)
```

Comment:
Perhaps we could use [`chooseInteger`](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/Test-QuickCheck.html#v:chooseInteger) here, as the QC documentation claims that this is faster.